import/src/test/java/com/google/cloud/healthcare/imaging/dicomadapter/AttributesUtilTest.java

```diff
@@ -0,0 +1,220 @@
+        "instances?" + TagUtils.toHexString(Tag.ModalitiesInStudy) + "=CS&");
+    assertThat(results[2]).isEqualTo(
+        "instances?" + TagUtils.toHexString(Tag.ModalitiesInStudy) + "=CR&");
+  }
+
+  @Test
+  public void testJsonToAttributes_empty() throws Exception {
+    JSONObject jsonObj = new JSONObject();
+
+    Attributes attrs = AttributesUtil.jsonToAttributes(jsonObj);
+
+    assertThat(attrs).isEqualTo(new Attributes());
+  }
+
+  @Test
+  public void testJsonToAttributes_integerString() throws Exception {
+    // can be parsed as either string or int.
+    JSONObject jsonObj = new JSONObject("{\"" + TagUtils.toHexString(Tag.InstanceNumber)
+        + "\": {\"vr\": \"IS\",\"Value\": [1]}}");
+
+    Attributes attrs = AttributesUtil.jsonToAttributes(jsonObj);
+
+    Attributes expected = new Attributes();
+    expected.setInt(Tag.InstanceNumber, VR.IS, 1);
+    assertThat(attrs).isEqualTo(expected);
+  }
+
+  @Test
+  public void testJsonToAttributes_intType() throws Exception {
+    JSONObject jsonObj = new JSONObject("{\""
+        + TagUtils.toHexString(Tag.ConcatenationFrameOffsetNumber)
+        + "\": {\"vr\": \"UL\",\"Value\": [1, 1]}}");
+
+    Attributes attrs = AttributesUtil.jsonToAttributes(jsonObj);
+
+    Attributes expected = new Attributes();
+    expected.setInt(Tag.ConcatenationFrameOffsetNumber, VR.UL, 1, 1);
+    assertThat(attrs).isEqualTo(expected);
+  }
+
+  @Test(expected = NumberFormatException.class)
+  public void testJsonToAttributes_intType_notInt() throws Exception {
+    JSONObject jsonObj = new JSONObject("{\""
+        + TagUtils.toHexString(Tag.ConcatenationFrameOffsetNumber)
+        + "\": {\"vr\": \"UL\",\"Value\": [\"Gotcha\"]}}");
+
+    AttributesUtil.jsonToAttributes(jsonObj);
+  }
+
+  @Test
+  public void testJsonToAttributes_stringType() throws Exception {
+    JSONObject jsonObj = new JSONObject("{\""
+        + TagUtils.toHexString(Tag.QueryRetrieveLevel)
+        + "\": {\"vr\": \"CS\",\"Value\": [\"IMAGE\"]}}");
+
+    Attributes attrs = AttributesUtil.jsonToAttributes(jsonObj);
+
+    Attributes expected = new Attributes();
+    expected.setString(Tag.QueryRetrieveLevel, VR.CS, "IMAGE");
+    assertThat(attrs).isEqualTo(expected);
+  }
+
+  @Test
+  public void testJsonToAttributes_patientName() throws Exception {
+    JSONObject jsonObj = new JSONObject("{\""
+        + TagUtils.toHexString(Tag.PatientName)
+        + "\": {\"vr\": \"PN\",\"Value\": [{"
+        + "\"Alphabetic\": \"Yamada^Tarou\", "
+        + "\"Ideographic\": \"山田^太郎\", "
+        + "\"Phonetic\": \"やまだ^たろう\", "
+        + "}]}}");
+
+    Attributes attrs = AttributesUtil.jsonToAttributes(jsonObj);
+
+    Attributes expected = new Attributes();
+    expected.setString(Tag.PatientName, VR.PN, "Yamada^Tarou=山田^太郎=やまだ^たろう");
+    assertThat(attrs).isEqualTo(expected);
+  }
+
+  @Test
+  public void testJsonToAttributes_double() throws Exception {
+    // is it reliable (double equality comparison)?
```

Comment:
Given that it runs in the JVM I think double comparison will be repeatable here, so probably ok to remove this comment.

Reply:
Ok